src/plugins/console/server/routes/api/console/proxy/tests/params.test.ts

```diff
@@ -66,16 +74,16 @@ describe('Console Proxy Route', () => {
             getProxyRouteHandlerDeps({ proxy: { pathFilters: [/^\/foo\//, /^\/bar\//] } })
           );
 
-          (requestModule.proxyRequest as jest.Mock).mockResolvedValue(createResponseStub('foo'));
+          const mockResponse = opensearchServiceMock.createSuccessTransportRequestPromise('foo');
+          opensearchClient.asCurrentUser.transport.request.mockResolvedValueOnce(mockResponse);
 
           const { status } = await handler(
-            {} as any,
+            { core: requestHandlerContextMock, dataSource: {} as any },
             { headers: {}, query: { method: 'POST', path: '/foo/id' } } as any,
             opensearchDashboardsResponseFactory
           );
 
           expect(status).to.be(200);
-          expect((requestModule.proxyRequest as jest.Mock).mock.calls.length).to.be(1);
         });
       });
       describe('all match', () => {
```

Comment:
what is the difference between the one match and all match unit test here? seems like all they are testing is the status code.

Reply:
They are testing the `pathFilter` inputs, it's configured via `osd.yml -> console.pathFilter`.The inputs are different for these 2 cases.
- one match: `{ pathFilters: [/^\/foo\//, /^\/bar\//]`
- all match:  `{ pathFilters: [/^\/foo\//]`

It's testing this part of the code
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1e127b0a52925ee51d185798f00362d8eb59d2ce/src/plugins/console/server/routes/api/console/proxy/create_handler.ts#L134-L142

Reply:
thanks, that makes sense~